webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -4256,5 +4257,127 @@ private void setToolProperties(ToolConfiguration tool, String propList) {
             }
         }
     }
+    /**
+     * Copy the lessons pages from a site to another one. Optionally, the existing lessons pages can be deleted in destination.
+     * Links to resources are preserved and updated to the new site, but their copy is not included. copySiteContent can be used for that.
+     * Lessons pages copy does not work in copySiteContent method, neither in copySiteContentForTool.
+     * This logic could be added to copySiteContent to cover all cases, but unfortunately we were in a hurry.
+     *
+     * @param 	sessionid 			the id of a valid session
+     * @param 	sourcesiteid		the id of the source site
+     * @param 	destinationsiteid 	the id of the destiny site
+     * @param 	cleanUp				delete the existing Lessons pages (true) or preserve them (false)
+     * @return						success or exception message
+     */
+    public String copySiteLessonsPages(
+    		@WebParam(name = "sessionid", partName = "sessionid") @QueryParam("sessionid") String sessionid,
+    		@WebParam(name = "sourcesiteid", partName = "sourcesiteid") @QueryParam("sourcesiteid") String sourcesiteid,
+    		@WebParam(name = "destinationsiteid", partName = "destinationsiteid") @QueryParam("destinationsiteid") String destinationsiteid,
+    		@WebParam(name = "cleanUp", partName = "cleanUp") @QueryParam("cleanUp") boolean cleanUp)
+    {
+
+    	String toolid="sakai.lessonbuildertool";
+    	Session session = establishSession(sessionid);
+    	Set<String> toolsCopied = new HashSet<String>();
+    	Map transversalMap = new HashMap();
+
+    	try
+    	{
+    		//check if both sites exist
+    		Site site = siteService.getSite(sourcesiteid);
+    		site = siteService.getSite(destinationsiteid);
+
+    		// If not admin, check maintainer membership in the source site
+    		if (!securityService.isSuperUser(session.getUserId()) && !securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference()))
+    		{
+    			LOG.warn("WS copyResources(): Permission denied. Must be super user to copy a site in which you are not a maintainer.");
+    			throw new RuntimeException("WS copyResources(): Permission denied. Must be super user to copy a site in which you are not a maintainer.");
+    		}
+
+    		Map<String,String> entityMap = transferCopyEntitiesForLessonsPages(toolid, sourcesiteid, destinationsiteid, cleanUp);
+    		if(entityMap != null)
+    		{
+    			transversalMap.putAll(entityMap);
+    		}
+
+    		updateEntityReferences(toolid, sourcesiteid, transversalMap, site);
+    	}
+    	catch (Exception e)
+    	{
+    		LOG.error("WS copySiteContent(): " + e.getClass().getName() + " : " + e.getMessage(), e);
+    		return e.getClass().getName() + " : " + e.getMessage();
+    	}
+    	return "success";
+    }
 
+    protected Map transferCopyEntitiesForLessonsPages(String toolId, String fromContext, String toContext, boolean cleanup)
```

Comment:
Sorry for the late comment. I have been very busy with the new term start.

In SakaiScript.java, method copySiteContentForTool() relies on transferContentEntities(). Both methods seems to be quite obsolete respecting to their homonymous methods in SiteAction.java.

The rest of the tools seems to work fine with these methods, but Lessons failed. I had to choose between fixing the current methods to work also for Lessons or creating new methods only for Lessons. We were in a hurry with lots of complains, so I chose the second option.

You can see that transferCopyEntities() and transferCopyEntitiesForLessonsPages() are quite different. Maybe the first one could be replaced by the second without harm, but I just haven't tested it.

May I suggest merging this patch in case that anybody needs to copy Lessons pages and creating a new JIRA to unify both methods later?